### ISSUE TITLE
Updates (and other notifications) are screen-reader friendly

### ISSUE THREAD

**Issue by CareyJWilliams** [ChoicescriptIDE/main#40]
The notify prompts don't seem to 'grab' a screen-readers immediate attention. They can be navigated over, but I feel like most of these (errors, updates etc.) are items that should get guaranteed immediate attention.

Bootbox alerts (as used for providing names for projects etc) do seem to trigger immediate screen-reader response, but are a little too obtrusive for general use.

**Comment by rkingett** [comment]
I didn't even notice there were messages and stuff when I used it last 
time, but yes! This is essential! By the way, is there a screen reader 
compiled windows version for me to look at yet?

**Comment by CareyJWilliams** [comment]
Not yet, as this issue is blocking it. I would like the accessible binary
to support updates (that are accessible), so even if the compatibility
fallout is high, we can slowly patch away at the issues without rebuilding
and re-installing every time.

**Comment by rkingett** [comment]
Is the issue with popout dialogue boxes or with the main window? How 
difficult would it be to use the default windows dialogue boxes or 
system tray baloons? System tray baloons read fine with NVDA but if they 
are long messages that may not work.
What about having small sounds for error notifications and otherwise or 
is that a different thing? What kinds of notifications are causing the 
issue?

**Comment by CareyJWilliams** [comment]
Dialogue boxes work fine, but they are a little too obtrusive for the type
of notifications we're talking about. Subtle pop-ups, notifying of updates
or errors in a run. I'm reluctant to use integrated desktop notifications
as that won't translate to the web version of the app. I believe the
current notifications can be made to work, it's just finding the right way
to mark them up so that screen-readers notice them immediately.

**Comment by CareyJWilliams** [comment]
I've had some small success with the builtin MAC OS screen reader, although it looks like it will involve upgrading a dependency, which will require some work. What would be the general preference in terms of notification alert level? I believe there are two options: assertive and polite. Assertive seems more reliable but can interrupt current read-outs. Polite is supposed to wait and join the 'queue', but I've found it doesn't always seem to trigger.

**Comment by rkingett** [comment]
First, I would like to ask, is there a window or something where we can 
review past alerts? I think having assertive types for small things like 
spelling and stuff, people would need to ajust their screen readers and 
I'm not sure if people would want to ajust their settings so they can 
have a seemless experience. So, what about adding a notification sound 
effect with polite types of notification but then have a keyboard 
command that opens up a window with previous notifications? I am 
thinking of something like the windows action center in Windows 10. Not 
sure how that would work on Macs though because I use NVDA on windows.

**Comment by CareyJWilliams** [comment]
Any "issues" with your game, will fire an error alert and then get logged in the issues tab. Other alerts are merely informative in the current context and are not stored, as it would rarely make sense to do so. E.g. "game is now running", "file name already exists" and "update successful" etc. I'm not at all keen on the idea of adding sounds
 It's a lot of extra complexity and resource collection for what I can't imagine is much universal gain.

**Comment by rkingett** [comment]
Makes sense! I was actually thinking of one sound for all notifications. 
But giving this some more thought, I think that having aggressive 
notification screen reader alerts would be better.
